src/network/core/udp.cpp

```diff
@@ -137,7 +137,10 @@ void NetworkUDPSocketHandler::ReceivePackets()
 				Debug(net, 1, "Received a packet with mismatching size from {}", address.GetAddressAsString());
 				continue;
 			}
-			p.PrepareToRead();
+			if (!p.PrepareToRead()) {
+				Debug(net, 1, "Invalid packet received (too small / MAC validation failed)");
```

Comment:
I wonder if `MAC` is useful to the user. Maybe: `too small / encryption error` ? (also for the other instances of this message)